File: R/EndInteraction.R
EndInteraction = function(M, V, interaction){
  # Choose a type of interaction to ignore in your adjacency matrices
  # The interations are identified based on the index of species which has a 1
  # in V and the Transpose of V (in case of antagonism). 
  # These index are used to insert 0 on the matrices.
  #
  #Args:
  # M: mutualistic adjacency matrix
  # V: antagonistic adjacency matrix
  # interaction: type on interaction you want to ignore (mutualism MM, antagonism AA or cheaters AM)
  #
  #Return:
  # list of matrices M and V with 0 in interactions MM, AA or AM 
  #
  
  # check the interaction parameter
  if(interaction != "mutualism" & interaction != "antagonism" & interaction != "cheaters"){
    stop("Choose a valid interaction type")
  }
  
  # end interactions based on "interaction" parameter
  if(interaction == "antagonism"){
    index = V == t(V)
    V[index] = 0
    mats = list(M, V)
    return(mats)
  }
  
  if(interaction == "cheaters"){
    index = V != t(V)
    V[index] = 0
    M[index] = 0
    mats = list(M, V)
    return(mats)
  }
  
  else{
    index = M == t(M)
    M[index] = 0
    mats = list(M, V)
    return(mats)
  }
}

File: Python/EndInteraction.py
def EndInteraction(M, V, interaction):
    ''' Function to end some type of interaction in adjacency matrix M and V.
    If you choose "antagonism" the elements where V are equal to V transpose
    gonna be changed to zero. The same process goes to "mutualism" and
    cheaters.
    Args:
        M: Adjacency matrix of interactions with 0's
        V: Adjacency matrix of antagonistic interactions with 1's
        interaction: which interaction you wanna end
    Return:
        M and V matrices with a certain interaction ended'''

    # load modules
    import numpy as np

    # message error to a non possible interaction ex: competition
    if(interaction != "antagonism" 
        and interaction != "antagonism" and interaction != "antagonism"):
            print ("Choose a valid interactions type")

    # check and delete the interactions where the element is equal to his
    # transpose. The process is the same for all the interactions choosen
    if(interaction == "antagonism"):
        index = V == np.transpose(V)
        V[index] = 0
        return M, V

    if(interaction == "cheaters"):
        index = V != np.transpose(V)
        V[index] = 0
        M[index] = 0
        return M, V

    else:
        index = M == np.transpose(M) 
        M[index] = 0
        return M, V